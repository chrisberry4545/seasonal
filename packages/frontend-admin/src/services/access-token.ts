import { getStorage } from '../consts';

export const setAccessToken = (token: string) => {
  getStorage().accessToken = token;
};

export const getAccessTokenHeaders = (): HeadersInit => {
  const token = getStorage().accessToken;
  return token
  ? {
    Authorization: `Bearer ${token}`
  }
  : {};
};
