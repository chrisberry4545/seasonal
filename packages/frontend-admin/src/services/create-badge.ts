import { IBadge } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { BADGE_URL } from '../config';

export const createBadge = async (badge: IBadge): Promise<IBadge> =>
  makeAuthorizedRequest<IBadge>(BADGE_URL, {
    body: JSON.stringify(badge),
    method: 'POST'
  });
