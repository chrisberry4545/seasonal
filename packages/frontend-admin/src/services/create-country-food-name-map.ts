import { ICountryFoodNameMap } from '@chrisb-dev/seasonal-shared';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_FOOD_NAME_MAP_URL } from '../config';

export const createCountryFoodNameMap = async (
  country: ICountryFoodNameMap
): Promise<ICountryFoodNameMap> =>
  makeAuthorizedRequest<ICountryFoodNameMap>(COUNTRY_FOOD_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
