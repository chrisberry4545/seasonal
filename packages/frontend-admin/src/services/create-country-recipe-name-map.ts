import { ICountryRecipeNameMap } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_RECIPE_NAME_MAP_URL } from '../config';

export const createCountryRecipeNameMap = async (
  country: ICountryRecipeNameMap
): Promise<ICountryRecipeNameMap> =>
  makeAuthorizedRequest<ICountryRecipeNameMap>(COUNTRY_RECIPE_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
