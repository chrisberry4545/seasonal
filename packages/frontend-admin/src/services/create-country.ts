import { ICountry } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_URL } from '../config';

export const createCountry = async (country: ICountry): Promise<ICountry> =>
  makeAuthorizedRequest<ICountry>(COUNTRY_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
