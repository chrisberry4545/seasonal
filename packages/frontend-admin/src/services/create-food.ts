import { IFood } from '@chrisb-dev/seasonal-shared';
import { makeAuthorizedRequest } from './make-authorized-request';
import { FOOD_URL } from '../config';

export const createFood = async (food: IFood): Promise<IFood> =>
  makeAuthorizedRequest<IFood>(FOOD_URL, {
    body: JSON.stringify(food),
    method: 'POST'
  });
