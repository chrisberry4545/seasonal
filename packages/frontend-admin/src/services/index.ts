export * from './set-access-token';

export * from './create-badge';

export * from './create-country-food-name-map';

export * from './create-country-recipe-name-map';

export * from './create-country';

export * from './create-food';

export * from './create-recipe';

export * from './create-region-food-season-map';

export * from './create-region';

export * from './create-user';

export * from './delete-badge';

export * from './delete-country-food-name-map';

export * from './delete-country-recipe-name-map';

export * from './delete-country';

export * from './delete-food';

export * from './delete-recipe';

export * from './delete-region-food-season-map';

export * from './delete-region';

export * from './delete-user';

export * from './get-all-badges';

export * from './get-all-countries';

export * from './get-all-country-food-name-map';

export * from './get-all-country-recipe-name-map';

export * from './get-all-food';

export * from './get-all-recipes';

export * from './get-all-region-food-season-map';

export * from './get-all-regions';

export * from './get-all-users';

export * from './get-one-badge';

export * from './get-one-country';

export * from './get-one-country-food-name-map';

export * from './get-one-region-food-season-map';

export * from './get-one-country-recipe-name-map';

export * from './get-one-food';

export * from './get-one-recipe';

export * from './get-one-region';

export * from './get-one-user';

export * from './login-request';

export * from './make-authorized-request';

export * from './update-badge';

export * from './update-country';

export * from './update-country-food-name-map';

export * from './update-region-food-season-map';

export * from './update-country-recipe-name-map';

export * from './update-food';

export * from './update-recipe';

export * from './update-region';

export * from './update-user';
