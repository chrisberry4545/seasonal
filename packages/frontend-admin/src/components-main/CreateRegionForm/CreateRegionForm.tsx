import React, { FC } from 'react';
import { createRegion } from '../../services';
import { IDbRegion } from '@chrisb-dev/seasonal-shared';
import { BaseFormRegion } from '../BaseFormRegion/BaseFormRegion';
import { FORM_BUTTON_TEXT } from '../../consts';
import { LayoutWithTitle } from '../../components-layouts';

const createEmptyRegionItem = (): IDbRegion => ({
  code: '',
  countryId: '',
  isDisabled: false,
  lat: 0,
  lng: 0,
  name: ''
} as IDbRegion);

export const CreateRegionForm: FC<{}> = () =>
  <LayoutWithTitle title='Create Region'>
    <BaseFormRegion items={createEmptyRegionItem()}
      updateMethod={createRegion}
      buttonText={FORM_BUTTON_TEXT.CREATE} />
  </LayoutWithTitle>;
