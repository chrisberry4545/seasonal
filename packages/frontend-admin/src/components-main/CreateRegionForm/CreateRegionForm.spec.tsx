import React from 'react';
import { CreateRegionForm } from './CreateRegionForm';
import { shallow, ShallowWrapper } from 'enzyme';
import { IDbRegion } from '@chrisb-dev/seasonal-shared-models';
import * as services from '../../services';
import {
  BaseFormRegion
} from '../BaseFormRegion/BaseFormRegion';

jest.mock('../../services', () => ({
  createRegion: jest.fn()
}));
jest.mock('../../consts', () => ({
  FORM_BUTTON_TEXT: {
    CREATE: 'create'
  }
}));
jest.mock('../../components-layouts', () => ({
  LayoutWithTitle: () => 'LayoutWithTitle'
}));
jest.mock('../BaseFormRegion/BaseFormRegion', () => ({
  BaseFormRegion: () => 'BaseFormRegion'
}));

describe('<CreateRegionForm />', () => {
  let wrapper: ShallowWrapper;
  let mockCreate: jest.SpyInstance;

  beforeEach(() => {
    mockCreate = jest.spyOn(services, 'createRegion')
      .mockResolvedValue({} as IDbRegion);
    wrapper = shallow(
      <CreateRegionForm />
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('calls create on update', () => {
    const item = {} as IDbRegion;
    const update = wrapper.find(BaseFormRegion).props().updateMethod;
    if (update) {
      update(item);
    }
    expect(mockCreate).toHaveBeenCalledWith(item);
  });

});
