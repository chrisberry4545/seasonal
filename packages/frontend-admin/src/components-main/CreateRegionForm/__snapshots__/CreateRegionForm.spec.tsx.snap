// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRegionForm /> renders correctly 1`] = `
<LayoutWithTitle
  title="Create Region"
>
  <BaseFormRegion
    additionalConfig={
      Object {
        "code": Object {
          "type": "text",
          "validation": Array [
            [Function],
          ],
        },
        "id": Object {
          "type": "text",
          "validation": Array [
            [Function],
          ],
        },
      }
    }
    buttonText="create"
    items={
      Object {
        "countryId": undefined,
        "id": "",
        "isDisabled": false,
        "lat": 0,
        "lng": 0,
        "name": "",
      }
    }
    updateMethod={[MockFunction]}
  />
</LayoutWithTitle>
`;
